src/machine/image.c

```diff
@@ -0,0 +1,205 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-json.h"
+
+#include "fd-util.h"
+#include "fileio.h"
+#include "discover-image.h"
+#include "image.h"
+#include "io-util.h"
+#include "operation.h"
+#include "process-util.h"
+#include "string-table.h"
+#include "tmpfile-util.h"
+
+int clean_pool_read_first_entry(FILE *file, int child_error, sd_bus_error *error) {
+        _cleanup_free_ char *name = NULL;
+        bool success;
+        int r;
+
+        assert(file);
+
+        if (fseek(file, 0, SEEK_SET) < 0)
+                return log_debug_errno(errno, "Failed to seek to the beggining of tmp file: %m");
+
+        /* The resulting temporary file starts with a boolean value that indicates success or not. */
+        errno = 0;
+        size_t n = fread(&success, 1, sizeof(success), file);
+        if (n != sizeof(success)) {
+                log_debug_errno(errno_or_else(EIO), "Received unexpected amount of bytes: %m");
+                return child_error < 0 ? child_error : errno_or_else(EIO);
```

Comment:
Not a matter, but you can use RET_GATHER():
```
return RET_GATHER(child_error, log_debug_errno(errno_or_else(EIO), "..."));
```

Reply:
I'd prefer to keep as is. I think it's more simple to understand how it's written

Reply:
OK. No problem. Feel free to keep it as is.